src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -737,7 +737,7 @@ private async Task ScanTargetsAsync(TContext globalContext, IEnumerable<Skimmer<
 
                     DriverEventSource.Log.ReadArtifactStart(filePath);
                     // Reading the length property faults in the file contents.
-                    long sizeInBytes = perFileContext.CurrentTarget.Contents.Length;
+                    long sizeInBytes = perFileContext.CurrentTarget.SizeInBytes.Value;
```

Comment:
>SizeInBytes.Value [](http://example.com/codeflow?start=68&length=17)

...SizeInBytes?.Value ?? 0

SizeInBytes may be null in some weird condition.

Reply:
Yes, and we should not provoke those conditions. :) There is a unit test for this. It's non-obvious, I'll try to find it.